include/ncengine/graphics/SpotLight.h

```diff
@@ -0,0 +1,44 @@
+/**
+ * @file SpotLight.h
+ * @copyright Jaremie Romer and McCallister Romer 2024
+ */
+#pragma once
+
+#include "ncengine/ecs/Component.h"
+
+#include "ncmath/Vector.h"
+
+namespace nc::graphics
+{
+/** @brief Component representing a light pointing in a single direction with a radius for attenuation, and an inner and outer angle to specify both cone size and falloff. */
+struct SpotLight final : public ComponentBase
```

Comment:
Haven't looked myself yet, but is there any reason to derive from `ComponentBase`? IIRC, the only reason `PointLight` still does is for binary compatibility of the scene fragment in SS&DD.